.travis.yml

```diff
@@ -1,19 +1,26 @@
 language: go
 
+branches:
+  except:
+    - master
+  only:
+    - release
 go:
   - 1.13.x
 
 services:
   - docker
 
 script:
-  - make
+  - go test ./...
+  - curl -sfL https://git.io/goreleaser | sh -s -- check
 
 before_deploy:
-  - echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+  - test -n "$TRAVIS_TAG" && docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
```

Comment:
@sridhav  can we use existing way to pass docker password instead? Less chances to expose it in the logs

Reply:
Travis allows you to mask any environment variables.